src/app/rundown-view/components/minishelf/mini-shelf.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core'
+import { Segment } from '../../../core/models/segment'
+import { ConfigurationService } from '../../../shared/services/configuration-service'
+import { StudioConfiguration } from '../../../shared/services/studio-configuration'
+import { Tv2VideoClipAction } from '../../../shared/models/tv2-action'
+import { Subscription } from 'rxjs'
+import { ActionService } from '../../../shared/abstractions/action.service'
+import { MediaStateService } from '../../../shared/services/media-state.service'
+import { Media } from '../../../shared/services/media'
+
+@Component({
+  selector: 'sofie-mini-shelf',
+  styleUrls: ['./mini-shelf.component.scss'],
+  templateUrl: './mini-shelf.component.html',
+})
+export class MiniShelfComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() public segment: Segment
+  @Input() public videoClipAction: Tv2VideoClipAction | undefined
+
+  private readonly defaultAssetForThumbnail: string = 'assets/sofie-logo.svg'
+  protected mediaDuration: number = 0
+  private configurationServiceSubscription: Subscription
+  private studioConfiguration: StudioConfiguration | undefined
+
+  constructor(
+    private readonly actionService: ActionService,
+    private readonly configurationService: ConfigurationService,
+    private readonly mediaStateService: MediaStateService
+  ) {}
+
+  public ngOnInit(): void {
+    this.configurationServiceSubscription = this.configurationService.getStudioConfiguration().subscribe((studioConfiguration: StudioConfiguration) => {
+      this.studioConfiguration = studioConfiguration
+    })
+    void this.updateMediaDataDuration()
```

Comment:
`void` is in most cases a bad practice that hides an issue. In this case, if the promise returned by `updateMediaDataDuration` is rejected, then we won't catch it (besides from the console log). If this is a node process, then the unhandled rejected promise can crash the application.

I suggest to add a `.catch` that at least logs the error.

Reply:
Idea was to represent facts - i.e. if actual duration can not be determined - display --:-- which differs than 00:00 since latter is not true.

Reply:
> Idea was to represent facts - i.e. if actual duration can not be determined - display --:-- which differs than 00:00 since latter is not true.

I think this comment was to another thread?
But to answer it: I think the idea is nice. The easy solution would be to not display a duration if the duration is set to 0. But I don't mind it just saying 00:00.